include/deal.II/numerics/vector_tools.h

```diff
@@ -857,6 +857,47 @@ namespace VectorTools
                 const bool                          project_to_boundary_first = false);
 
   /**
+   * The same as above  for distributed Triangulations.
+   * The underlying MatrixFree implementation requires to specify the number
+   * of components and the degree of the used FiniteElement as additional
+   * template parameters.
+   * Currently, the implementation only allows for @p enforce_zero_boundary
+   * and @p project_to_boundary_first to be <tt>false</tt>.
+   * @p vec_result is assumed to not have a ghost elements.
+   */
+  template <int dim, typename VectorType, int spacedim, int components, int fe_degree>
```

Comment:
I think we should remove both template arguments and provide pre-compiled paths. Regarding the number of components, if we pre-compile up to `4` components (3+1 for 3D), the rest can be treated with generic code in FEEvaluation that is upcoming.